From: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date: Mon, 27 Mar 2017 14:18:29 +0200
Subject: khtml: Delete pointer _before_ returning.

Fixup to commit 947ffe60 ("backport 02c3192 - Patch some memory leaks"),
which backported part of the fix in the wrong order, as it was first
returning and then deleting a pointer.

Noticed by Matthew Rezny <rezny@FreeBSD.org>.
---
 khtml/ecma/kjs_html.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/khtml/ecma/kjs_html.cpp b/khtml/ecma/kjs_html.cpp
index 71e6f6bc46..535b1e7390 100644
--- a/khtml/ecma/kjs_html.cpp
+++ b/khtml/ecma/kjs_html.cpp
@@ -3364,8 +3364,8 @@ void KJS::HTMLSelectCollection::put(ExecState *exec, const Identifier &propertyN
         static_cast<DOM::HTMLElementImpl*>(element->document()->createElement("OPTION")),
         before, exception);
       if (exception.triggered()) {
-          return;
           delete option;
+          return;
       }
     }
     // replace an existing entry ?
